#!/usr/bin/env bash

# Purpose: Evaluate the quality of genome assemblies using MERQURY to assess k-mer
# accuracy and completeness of genome assemblies

# Tool:
# MERQURY v1.3

# Inputs:
# - `*.gz`: Raw genome read files for the Edi-0 sample
# - `*.fasta`: Assembly files generated by different assemblers

# Outputs:
# - `merquryEdi-0/`: MERQURY results

###############################################################################

#SBATCH --cpus-per-task=16
#SBATCH --time=1-00:00:00
#SBATCH --mem=64G
#SBATCH --partition=pibu_el8
#SBATCH --job-name=merqury
#SBATCH --mail-user=ana.castromarquez@students.unibe.ch
#SBATCH --mail-type=begin,end
#SBATCH --output=/data/users/acastro/genome_assembly_annotation/log/output_merqury_%j.o
#SBATCH --error=/data/users/acastro/genome_assembly_annotation/log/error_merqury_%j.e

# Go to working directory
cd /data/users/$USER/genome_assembly_annotation

# Input directories for assemblies
DATA_DIR_EDI=/data/users/$USER/genome_assembly_annotation/raw_data/Edi-0
DIR_FLYE_EDI=/data/users/$USER/genome_assembly_annotation/analysis/flyeEdi-0
DIR_HIFIASM_EDI=/data/users/$USER/genome_assembly_annotation/analysis/hifiasmEdi-0
DIR_LJA_EDI=/data/users/$USER/genome_assembly_annotation/analysis/ljaEdi-0

# Output directory for MERQURY results
DIR_MERQURY_EDI=/data/users/$USER/genome_assembly_annotation/analysis/merquryEdi-0

# Create directories to save results
mkdir --parents $DIR_MERQURY_EDI

# Set MERQURY path
export MERQURY="/usr/local/share/merqury"

# Create Meryl databases for genome
apptainer exec /containers/apptainer/merqury_1.3.sif \
    meryl \
    k=21 \
    count \
    $DATA_DIR_EDI/*.gz \
    output $DIR_MERQURY_EDI/genome.meryl

# Run MERQURY analysis for Flye assembly
apptainer exec /containers/apptainer/merqury_1.3.sif \
    $MERQURY/merqury.sh \
    $DIR_MERQURY_EDI/genome.meryl \
    $DIR_FLYE_EDI/assembly.fasta \
    merqury_flye

# Run MERQURY analysis for LJA assembly
apptainer exec /containers/apptainer/merqury_1.3.sif \
    $MERQURY/merqury.sh \
    $DIR_MERQURY_EDI/genome.meryl \
    $DIR_LJA_EDI/assembly.fasta \
    merqury_lja

# Run MERQURY analysis for Hifiasm assembly
apptainer exec /containers/apptainer/merqury_1.3.sif \
    $MERQURY/merqury.sh \
    $DIR_MERQURY_EDI/genome.meryl \
    $DIR_HIFIASM_EDI/hifiasm_output.bp.p_utg.fa \
    merqury_hifiasm
