#!/usr/bin/env bash

# Purpose: Compare genome assemblies against a reference genome using NUCMER,
# and generate plots to visualize the alignments

# Tool:
# MUMmer v4x

# Inputs:
# - `*.TAIR10.dna.toplevel.fa`: Reference genome file
# - `*.fasta`: Assembly files generated by different assemblers

### Outputs:
# - `nucmer/`:
#   - *.png: MUMmer plots

###############################################################################

#SBATCH --cpus-per-task=11
#SBATCH --time=1-00:00:00
#SBATCH --mem=16G
#SBATCH --partition=pibu_el8
#SBATCH --job-name=nucmer
#SBATCH --mail-user=ana.castromarquez@students.unibe.ch
#SBATCH --mail-type=begin,end
#SBATCH --output=/data/users/acastro/genome_assembly_annotation/log/output_nucmer_%j.o
#SBATCH --error=/data/users/acastro/genome_assembly_annotation/log/error_nucmer_%j.e

# Go to working directory
cd /data/users/$USER/genome_assembly_annotation

# Input directory for the reference genome
DATA_DIR_EDI_REF=/data/users/$USER/genome_assembly_annotation/raw_data/references

# Input directories for assemblies
DIR_FLYE_EDI=/data/users/$USER/genome_assembly_annotation/analysis/flyeEdi-0
DIR_HIFIASM_EDI=/data/users/$USER/genome_assembly_annotation/analysis/hifiasmEdi-0
DIR_LJA_EDI=/data/users/acastro/genome_assembly_annotation/analysis/ljaEdi-0

# Output directory for MERQURY results
DIR_NUCMER=/data/users/$USER/genome_assembly_annotation/analysis/nucmer

# Create directory to save results
mkdir --parents $DIR_NUCMER

# Run NUCMER for Flye assembly comparison against the reference genome
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    nucmer \
    --prefix=$DIR_NUCMER/flye \
    --mincluster=100 \
    $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    $DIR_FLYE_EDI/assembly.fasta

# Run NUCMER for Hifiasm assembly comparison against the reference genome
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    nucmer \
    --prefix=$DIR_NUCMER/hifiasm \
    --mincluster=100 \
    $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    $DIR_HIFIASM_EDI/hifiasm_output.bp.p_utg.fa

# Run NUCMER for LJA assembly comparison against the reference genome
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    nucmer \
    --prefix=$DIR_NUCMER/lja \
    --mincluster=100 \
    $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    $DIR_LJA_EDI/assembly.fasta

# Plots for Flye assembly results
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    mummerplot \
    --prefix=$DIR_NUCMER/flye \
    -Rfile $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    -Qfile $DIR_FLYE_EDI/assembly.fasta \
    -breaklen 1000 \
    --filter \
    -t png \
    --large \
    --layout \
    --fat \
    $DIR_NUCMER/flye.delta

# Plots for Hifiasm assembly results
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    mummerplot \
    --prefix=$DIR_NUCMER/hifiasm \
    -Rfile $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    -Qfile $DIR_HIFIASM_EDI/hifiasm_output.bp.p_utg.fa \
    -breaklen 1000 \
    --filter \
    -t png \
    --large \
    --layout \
    --fat \
    $DIR_NUCMER/hifiasm.delta

# Plots for LJA assembly results
apptainer exec /containers/apptainer/mummer4_gnuplot.sif \
    mummerplot \
    --prefix=$DIR_NUCMER/lja \
    -Rfile $DATA_DIR_EDI_REF/Arabidopsis_thaliana.TAIR10.dna.toplevel.fa \
    -Qfile $DIR_LJA_EDI/assembly.fasta \
    -breaklen 1000 \
    --filter \
    -t png \
    --large \
    --layout \
    --fat \
    $DIR_NUCMER/lja.delta
